File: app/src/main/java/com/leodeleon/popmovies/di/modules/FragmentBuildersModule.java
package com.leodeleon.popmovies.di.modules;

import com.leodeleon.popmovies.feature.view.DetailFragment;
import com.leodeleon.popmovies.feature.view.MoviesFragment;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module public abstract class FragmentBuildersModule {
  @ContributesAndroidInjector
  abstract DetailFragment detailFragment();

  @ContributesAndroidInjector
  abstract MoviesFragment moviesFragment();
}

File: app/src/main/java/com/leodeleon/popmovies/di/modules/FragmentBuildersModule.kt
package com.leodeleon.popmovies.di.modules

import com.leodeleon.popmovies.feature.view.DetailFragment
import com.leodeleon.popmovies.feature.view.MoviesFragment
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module abstract class FragmentBuildersModule {
  @ContributesAndroidInjector
  internal abstract fun detailFragment(): DetailFragment

  @ContributesAndroidInjector
  internal abstract fun moviesFragment(): MoviesFragment
}
